k256/src/arithmetic/scalar.rs

```diff
@@ -227,6 +238,82 @@ impl Scalar {
     }
 }
 
+impl ff::Field for Scalar {
+    fn random<R: RngCore + ?Sized>(rng: &mut R) -> Self {
+        Scalar::generate_vartime(rng)
+    }
+
+    fn zero() -> Self {
+        Scalar::zero()
+    }
+
+    fn one() -> Self {
+        Scalar::one()
+    }
+
+    fn is_zero(&self) -> bool {
+        self.0.is_zero().into()
+    }
+
+    #[must_use]
+    fn square(&self) -> Self {
+        Scalar::square(self)
+    }
+
+    #[must_use]
+    fn double(&self) -> Self {
+        todo!();
+    }
+
+    fn invert(&self) -> CtOption<Self> {
+        Scalar::invert(self)
+    }
+
+    fn sqrt(&self) -> CtOption<Self> {
+        todo!();
+    }
```

Comment:
Stub!

Reply:
It seems there's some code we can repurpose from the `p256` crate to impl this, but in the meantime I've more clearly marked it's a stub and would like to address that in a follow-up PR.